Framework/DataHandling/test/LoadISISNexusTest.h

```diff
@@ -769,6 +772,316 @@ class LoadISISNexusTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS("PNR",
                      extractStringLog(*secondMatrixWS, "measurement_type"));
   }
+
+  void test_that_non_contiguous_data_loads_for_excluded_monitors() {
+    /*
+    Monitors can be found at detID: 145, 146, 147, 148
+    */
+    // Arrange
+    Mantid::API::FrameworkManager::Instance();
+    LoadISISNexus2 ld;
+    ld.initialize();
+    ld.setPropertyValue("Filename", "INS09161.nxs");
+    ld.setPropertyValue("OutputWorkspace", "outWS");
+    ld.setPropertyValue("LoadMonitors", "0" /* exclude monitors*/);
+    // Act
+    TS_ASSERT_THROWS_NOTHING(ld.execute());
+    TS_ASSERT(ld.isExecuted());
+
+    // Assert
+    MatrixWorkspace_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("outWS");
+    TSM_ASSERT_EQUALS("Should have 17036 bins", ws->blocksize(), 17036);
+    TSM_ASSERT_EQUALS("Should have 168 detectors (no monitors)",
+                      ws->getNumberHistograms(), 168);
+
+    std::vector<Mantid::detid_t> monitorDetIDs{145, 146, 147, 148};
+    std::vector<Mantid::detid_t> neighborsToCheck{140, 141, 142, 143,
+                                                  144, 149, 150, 151};
+    auto detIDtoWSIndexMap = ws->getDetectorIDToWorkspaceIndexMap();
+
+    // Check monitors are not in workspace
+    for (const auto &monitorDetID : monitorDetIDs) {
+      TSM_ASSERT("Should not be in the detID2WSIndexMap.",
+                 detIDtoWSIndexMap.count(monitorDetID) == 0);
+    }
+    for (const auto &neighborToCheck : neighborsToCheck) {
+      TSM_ASSERT("Should be in the detID2WSIndexMap.",
+                 detIDtoWSIndexMap.count(neighborToCheck) == 1);
+    }
+
+    // Check some of the data
+    double delta = 1e-6;
+    TS_ASSERT_DELTA(ws->readY(142)[0], 0.0, delta);
```

Comment:
The tests here and below are really nice and thorough for each of the cases regarding monitor data location. While we have the check that the monitor IDs have been loaded into the `detIDtoWSIndexMap` what I think we are missing is the check for the value of `getSpectrumNo()` and `getDetectorIDs()` for the spectra tested below.

I think we need this to be sure that the correct spectrum number/ID has been assigned to each histogram. The map only checks that they are there. I this needs to be checked for each of these test methods.